### PR TITLE
Fix duplicate message rendering

### PR DESCRIPTION
**Pull request by RafaelJCamara**
## Description
The line that was causing the duplication was:
`0 => new(EventData.PayloadNames[0], EventData.Payload[0])`

So I got rid of it.

Also, I assumed that this line was a part of the `Count` property, which was 5 before, and I reduced it to 4 now.

Based on this, I also slide down the indices of the index switch.

_**Before**_:

_**After**_:
2024-11-27_15h32_20

Fixes **_partially_** #2967 

## Checklist

- Is this feature complete?
  - [X] Yes. Ready to ship.
  - [ ] No. Follow-up changes expected.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [X] Yes
  - [ ] No
- Did you add public API?
  - [ ] Yes
    - If yes, did you have an API Review for it?
      - [ ] Yes
      - [ ] No
    - Did you add `<remarks />` and `<code />` elements on your triple slash comments?
      - [ ] Yes
      - [ ] No
  - [X] No
- Does the change make any security assumptions or guarantees?
  - [ ] Yes
    - If yes, have you done a threat model and had a security review?
      - [ ] Yes
      - [ ] No
  - [X] No
- Does the change require an update in our Aspire docs?
  - [ ] Yes
    - Is this introducing a breaking change?
      - [ ] Yes
        - Link to aspire-docs issue (please use this [`breaking-change` template](https://github.com/dotnet/docs-aspire/issues/new?template=04-breaking-change.yml)):
      - [ ] No
        - Link to aspire-docs issue (please use this [`doc-idea` template](https://github.com/dotnet/docs-aspire/issues/new?template=02-docs-request.yml)):
  - [X] No

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/6827)

### CODE REVIEW

**Comment by RafaelJCamara** [comment]
> @RafaelJCamara please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @dotnet-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @dotnet-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @dotnet-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@dotnet-policy-service agree

File: src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs [hunk 1]
```diff
@@ -139,24 +139,23 @@ IEnumerator IEnumerable.GetEnumerator()
 
                 return index switch
                 {
-                    0 => new(EventData.PayloadNames[0], EventData.Payload[0]),
```
Comment (eerhardt) [inline src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs:142]: I don't think we should be making this change. The `Message` property is separate from the `exception.message`. See the following code in the RabbitMQ.Client library:

https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/58ac94966d12ffbb88ccd03ea085b3516a544321/projects/RabbitMQ.Client/Impl/AutorecoveringConnection.Recovery.cs#L118
https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/58ac94966d12ffbb88ccd03ea085b3516a544321/projects/RabbitMQ.Client/Impl/AutorecoveringConnection.Recovery.cs#L159
https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/58ac94966d12ffbb88ccd03ea085b3516a544321/projects/RabbitMQ.Client/Impl/AutorecoveringConnection.Recovery.cs#L219

The messages can and will be different. By not emitting the `Message` property, we will be losing information.

Comment (RafaelJCamara) [reply src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs:142]: > Also, we need to update the Info/Warn messages (i.e. the EventSourceEvent struct above) as well. I don't think we need a state object at all in that case.

Do you think we should do this in the scope of this issue? I think we could skip this for now.

Comment (eerhardt) [reply src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs:142]: It's the same problem. Just a different manifestation of it - Error vs. Warn/Info.

Comment (eerhardt) [reply src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs:142]: image

Comment (RafaelJCamara) [reply src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs:142]: Corrected on the latest commit!

**Comment by RafaelJCamara** [comment]
This is how it looks like now for Info/Warning:
_**Before**_:
image


_**After**_:
info_rmq_after

File: src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ private void LogEvent(EventWrittenEventArgs eventData)
                 (eventData.EventId == 1 && eventData.EventName == "Info") ||
                 (eventData.EventId == 2 && eventData.EventName == "Warn"));
 
-            _logger.Log(level, eventId, new EventSourceEvent(eventData), null, s_formatEvent);
+            _logger.Log(level, eventId, s_formatEvent(new EventSourceEvent(eventData), null));
```
Comment (eerhardt) [inline src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs:53]: Please revert this change. This is now creating a closure (i.e. an allocation) on every log message.

Comment (eerhardt) [reply src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs:53]: Instead, you want to change `EventSourceEvent` to return `0` from its Count and its indexer.

File: src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs [hunk 3]
```diff
@@ -136,30 +102,20 @@ IEnumerator IEnumerable.GetEnumerator()
                 Debug.Assert(EventData.PayloadNames[1] == "ex");
 
                 ArgumentOutOfRangeException.ThrowIfGreaterThanOrEqual(index, 5);
```
Comment (eerhardt) [inline src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs:104]:
```suggestion
                ArgumentOutOfRangeException.ThrowIfGreaterThanOrEqual(index, Count);
```

File: src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs [hunk 4]
```diff
@@ -136,30 +102,20 @@ IEnumerator IEnumerable.GetEnumerator()
...
+                Debug.Assert(index >= 0 && index <= 3);
+                Debug.Assert(EventData.Payload?.Count == 2);
```
Comment (eerhardt) [inline src/Components/Aspire.RabbitMQ.Client/RabbitMQEventSourceLogForwarder.cs:106]:
```suggestion
```

These are duplicated and don't add value.

**Review by eerhardt — APPROVED** [review]
LGTM. Thank you for the contribution!
